### PR TITLE
Public async operations.

### PR DESCRIPTION
**Pull request by vcsjones**
For consumers of this type that can do async, let them do async.

### CODE REVIEW

File: RSAKeyVaultProvider/RSAKeyVault.cs [hunk 1]
```diff
@@ -67,15 +72,20 @@ protected override byte[] HashData(Stream data, HashAlgorithmName hashAlgorithm)
                 return digestAlgorithm.ComputeHash(data);
             }
         }
-        
+
         public override byte[] Decrypt(byte[] data, RSAEncryptionPadding padding)
         {
+                // Key Vault's API is known to use CA(false) everywhere. This should not deadlock.
+            return DecryptAsync(data, padding).GetAwaiter().GetResult();
```
Comment (onovotny) [inline RSAKeyVaultProvider/RSAKeyVault.cs]: Does this need a CA(false) here?

File: RSAKeyVaultProvider/RSAKeyVault.cs [hunk 2]
```diff
@@ -23,16 +23,21 @@ public RSAKeyVault(KeyVaultContext context)
 
         public override byte[] SignHash(byte[] hash, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding)
         {
+            // Key Vault's API is known to use CA(false) everywhere. This should not deadlock.
+            return SignHashAsync(hash, hashAlgorithm, padding).GetAwaiter().GetResult();
+        }
```
Comment (onovotny) [inline RSAKeyVaultProvider/RSAKeyVault.cs]: Does this need a CA(false) here?

Comment (vcsjones) [inline RSAKeyVaultProvider/RSAKeyVault.cs]: Not sure... it wasn't there originally. I just extracted the async parts and made them public.

Comment (vcsjones) [inline RSAKeyVaultProvider/RSAKeyVault.cs]: Oh wait. They were there.

**Comment by onovotny** [comment]
I didn't originally have this as I assumed that anyone who'd be "Key Vault aware" would just call the Key Vault API directly. I imagine the primary consumer of this type would be to plug-in to places that only know of the RSA or AsymmetricAlgorithm and thus only use the sync api's.

**Comment by vcsjones** [comment]
I think it has use aside from that. This "speaks" other .NET APIs, like HashAlgorirthmName instead of JWK identifiers. I can pull this into OOST now and dump some code.

**Comment by onovotny** [comment]
will update the nuget shortly then

**Comment by onovotny** [comment]
Updated to 1.0.1 with these changes

**Pull request merged**